### DISCUSSION TITLE
Autograding Reporter Seems Broken

### DISCUSSION THREAD

**Discussion by MisterBrash** [community/community#116650] [GitHub Education]
### Hi everyone,

The autograding feature took me a _really_ long time to learn (teaching JavaScript to HS students). I would love it if you could put all the tests in one file and have the reporter score based on the outcome of ALL the tests. But I digress, let me get to my original issue...

I have a task for my class with multiple test files (one for each function). Each test file has multiple tests (maybe 3-6 tests per function). 

If a test fails _some_ of the tests in a file, the Autograding Reporter still gives full points to the student.

Example:

Test file "area.test.js" contains 5 tests. The student submits to GitHub and the autograder does its job. The student passes 1/5 tests (4 fail). The Autograding Report gives the student full marks and says they passed the test.

Any help or guidance is appreciated.

**Comment by samus-aran** [comment]
Hi @MisterBrash 👋 

Thank you for raising this. Are you able to share with us the details of your assignment set up and a student repo so we can investigate? That would be much appreciated 👍

**Reply by MisterBrash** [reply]
Yes, I'd be happy to. I will make a new one and share the link as soon as the "500 Internal Error" stops happening on making new assignments.

**Reply by MisterBrash** [reply]
Ok, I think I figured out half my problem.
In short:
- I'm new to unit tests and Jest. High school teacher, not from industry, would love some very specific documentation on how to use this stuff.
- I was using `npm test` and was installing the `jest` package. This seemed to cause the issues, somehow. Just now I tested it by using `npm i -g jest-cli` and running `jest <filename>` and it works.

So I think this can be closed and written up as new user error. n00b

I'm hoping that there will be very clear documentation in the future because this has been _weeks_ of testing stuff and trying this and that and the other thing.

Also, it would be **amazing** if the points awarded were based on how many tests passed in a single test file. For example, if the file `area.test.js` has 10 tests in it and the GH Classroom "test" provides 10 points, if the student work fails 2 of the tests, they should receive 8/10 points.

But now I've derailed my own thread.

Cheers.
